server/api/public.py

```diff
@@ -42,15 +47,106 @@
+) -> sampler_contest.Contest:
+    validate(compute_sample_sizes_input, COMPUTE_SAMPLE_SIZES_INPUT_SCHEMA)
+
+    election_results = compute_sample_sizes_input["electionResults"]
+    candidates = election_results["candidates"]
+    num_winners = election_results["numWinners"]
+    total_ballots_cast = election_results["totalBallotsCast"]
+
+    if all(candidate["votes"] == 0 for candidate in candidates):
+        raise Conflict("At least 1 candidate must have greater than 0 votes")
+
+    if num_winners >= len(candidates):
+        raise Conflict("Number of winners must be less than number of candidates")
+
+    candidate_votes_sum = sum(candidate["votes"] for candidate in candidates)
+    if total_ballots_cast < candidate_votes_sum:
+        raise Conflict("Total ballots cast cannot be less than sum of candidate votes")
+
+    contest = Contest(
+        choices=[
+            ContestChoice(
+                id=f"candidate-{i}",
+                num_votes=candidate["votes"]
+                + 0.01  # Safeguard against math errors involving 0s
+                if candidate["votes"] == 0
+                else candidate["votes"],
+            )
+            for i, candidate in enumerate(candidates)
+        ],
+        num_winners=num_winners,
+        total_ballots_cast=total_ballots_cast,
+        votes_allowed=num_winners,
+    )
+    contest_for_sampler: sampler_contest.Contest = sampler_contest.from_db_contest(
+        contest
+    )
+    return contest_for_sampler
+
+
+def is_full_hand_tally_required(
+    contest: sampler_contest.Contest, risk_limit_percentage: int
+) -> bool:
+    is_tie = contest.diluted_margin == 0
+    return is_tie or risk_limit_percentage == 0
+
+
+def compute_ballot_comparison_sample_size(
+    contest: sampler_contest.Contest, risk_limit_percentage: int
+) -> int:
+    if is_full_hand_tally_required(contest, risk_limit_percentage):
+        return contest.ballots
+
+    sample_size = supersimple.get_sample_sizes(risk_limit_percentage, contest, None)
+    return min(sample_size, contest.ballots)
+
+
+def compute_ballot_polling_sample_size(
+    contest: sampler_contest.Contest, risk_limit_percentage: int
+) -> int:
+    if is_full_hand_tally_required(contest, risk_limit_percentage):
+        return contest.ballots
+
+    sample_size_options = bravo.get_sample_size(
+        risk_limit_percentage, contest, None, None
+    )
+    sample_size: int = sample_size_options.get(
+        "0.9", sample_size_options.get("asn", {"size": contest.ballots}),
+    )["size"]
+    return min(sample_size, contest.ballots)
+
+
+def compute_batch_comparison_sample_size(
+    contest: sampler_contest.Contest, risk_limit_percentage: int
+) -> int:
+    if is_full_hand_tally_required(contest, risk_limit_percentage):
+        return contest.ballots
+
+    sample_size = int(
+        math.ceil(-1 * math.log(risk_limit_percentage / 100) / contest.diluted_margin)
+        + 2
+    )
+    return min(sample_size, contest.ballots)
```

Comment:
Interesting question here: What should the max possible number of batches for a batch comparison audit be? For audit methods where the unit is ballots, it's clear that the max should be all ballots. For batches, we don't actually know how many total batches there are, and I'm not sure what the average number of ballots per batch is.

There's also a general design question here: What should the UI display when we want to recommend a full hand tally?
- The total number of ballots/batches? (What this PR currently does)
- "All ballots" / "All batches"?
- "Full hand tally"?

Reply:
I think I like "Full hand tally" the most, but would love @ginvdr's thoughts.

Reply:
I lean towards "Full hand tally" as well

Reply:
Voilà! https://github.com/votingworks/arlo/pull/1600/commits/7e1c9503bcc1613d5fc0e94a5c1a91760272b770

<img width="660" alt="full-hand-tally" src="https://user-images.githubusercontent.com/12616928/192889064-8d3de46c-56fd-46da-839d-87972dd2c5ab.png">